commit 4fc04b92fcafa7c8719e8f3c4f9fe5771dc78287
Author: popcornmix <popcornmix@gmail.com>
Date:   Fri Apr 27 00:30:15 2012 +0100

    Fix (hopefully) for DWC_MEMCPY kernel panics. Thanks to Naren Sankar for finding this

Index: linux-3.2.46/drivers/usb/host/dwc_otg/dwc_otg_hcd.c
===================================================================
--- linux-3.2.46.orig/drivers/usb/host/dwc_otg/dwc_otg_hcd.c	2013-07-26 19:31:23.000000000 +0000
+++ linux-3.2.46/drivers/usb/host/dwc_otg/dwc_otg_hcd.c	2013-07-26 19:32:45.000000000 +0000
@@ -909,6 +909,10 @@
 		return 0;
 #endif
 
+	if (((urb->actual_length < 0) || (urb->actual_length > urb->length)) && !dwc_otg_hcd_is_pipe_in(&urb->pipe_info))
+		urb->actual_length = urb->length;
+
+
 	hc = DWC_CIRCLEQ_FIRST(&hcd->free_hc_list);
 
 	/* Remove the host channel from the free list. */
Index: linux-3.2.46/drivers/usb/host/dwc_otg/dwc_otg_hcd_linux.c
===================================================================
--- linux-3.2.46.orig/drivers/usb/host/dwc_otg/dwc_otg_hcd_linux.c	2013-07-26 19:31:23.000000000 +0000
+++ linux-3.2.46/drivers/usb/host/dwc_otg/dwc_otg_hcd_linux.c	2013-07-26 19:32:45.000000000 +0000
@@ -668,6 +668,9 @@
 					    urb->number_of_packets,
 					    mem_flags == GFP_ATOMIC ? 1 : 0);
 
+	if(dwc_otg_urb == NULL)
+		return -ENOMEM;
+
         urb->hcpriv = dwc_otg_urb;
         
 	dwc_otg_hcd_urb_set_pipeinfo(dwc_otg_urb, usb_pipedevice(urb->pipe),
